File: app/src/main/java/mal/art/shopin/ViewModel/ProductViewModel.java
package mal.art.shopin.ViewModel;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import java.util.List;
import mal.art.shopin.Model.Product;
import mal.art.shopin.Repository.ProductRepository;

public class ProductViewModel extends AndroidViewModel {

  private ProductRepository repository;

  private LiveData<List<Product>> allProducts;

  public ProductViewModel(@NonNull Application application) {
    super(application);
    repository = new ProductRepository(application);
    allProducts = repository.getAllProducts();
  }

  public LiveData<List<Product>> getAllProducts() {
    return allProducts;
  }

  public void insert(Product product) {
    repository.insert(product);
  }
}

File: app/src/main/java/mal/art/shopin/viewModel/ProductViewModel.kt
package mal.art.shopin.viewModel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.google.firebase.firestore.CollectionReference
import mal.art.shopin.model.ProductModel
import mal.art.shopin.repository.ProductRepository

class ProductViewModel(application: Application) : AndroidViewModel(application) {

  private val repository: ProductRepository = ProductRepository(application)

  val allProducts: LiveData<List<ProductModel?>?>?
    get() = repository.allProducts

  fun insert(productModel: ProductModel?) {
    repository.insert(productModel)
    println("produkt wsadzony: $productModel")
  }

  fun testFirestore(productModel: ProductModel) {
    repository.testFirestore(productModel)
    println("produktFirestore: $productModel")
  }

  fun testFirestore2() {
    repository.testFirestore2()
  }

  fun getShoppingListFirestoreReference(): CollectionReference {
    return repository.getShoppingListFirestoreReference()
  }

}